src/mlpack/core/data/load_utilities.hpp

```diff
@@ -0,0 +1,490 @@
+
+  //! Modify the ImageInfo.
+  ImageInfo& ImageInfos() { return imgInfo; }
+
+  //! Get the FileType.
+  const DatasetInfo& Mapper() const { return mapper; }
+
+  //! Modify the DatasetMapper.
+  DatasetInfo & Mapper() { return mapper; }
+
+  //! Get if we are loading an image.
+  const std::string& ObjectName() const { return objectName; }
+
+  //! Modify if we are loading an image.
+  std::string& ObjectName() { return objectName; }
+
+  /**
+   * Given a file type, return a logical name corresponding to that file type.
+   */
+  const std::string FileTypeToString() const
+  {
+    switch (fileFormat)
+    {
+      case FileType::CSVASCII:    return "CSV data";
+      case FileType::RawASCII:    return "raw ASCII formatted data";
+      case FileType::RawBinary:   return "raw binary formatted data";
+      case FileType::ArmaASCII:   return "Armadillo ASCII formatted data";
+      case FileType::ArmaBinary:  return "Armadillo binary formatted data";
+      case FileType::PGMBinary:   return "PGM data";
+      case FileType::HDF5Binary:  return "HDF5 data";
+      case FileType::CoordASCII:  return "ASCII formatted sparse coordinate data";
+      default:                    return "";
+    }
+  }
+
+ private:
+
+  bool fatal;
+  bool hasHeaders;
+  bool noTranspose;
+  bool semiColon;
+  bool missingToNan;
+  bool categorical;
+  bool image;
+  bool model;
+  FileType fileFormat;
+  format dataFormat;
+  std::string objectName;
+  arma::field<std::string> headers;
+  ImageInfo imgInfo;
+  DatasetInfo mapper;
+};
+
+inline DataOptions operator|(const DataOptions& a, const DataOptions& b)
+{
+  DataOptions output;
+  output.Fatal() = a.Fatal() | b.Fatal();
+  output.NoTranspose() = a.NoTranspose() | b.NoTranspose();
+  output.SemiColon() = a.SemiColon() | b.SemiColon();
+  output.MissingToNan() = a.MissingToNan() | b.MissingToNan();
+  output.Categorical() = a.Categorical() | b.Categorical();
+  output.Image() = a.Image() | b.Image();
+  output.Model() = a.Model() | b.Model();
+
+  if (a.FileFormat() == FileType::FileTypeUnknown)
+  {
+    output.FileFormat() = b.FileFormat();
+    return output;
+  }
+  else if (b.FileFormat() == FileType::FileTypeUnknown)
+  {
+    output.FileFormat() = a.FileFormat();
+    return output;
+  }
+
+  if (a.FileFormat() != b.FileFormat())
+    throw std::runtime_error("File formats don't match!");
+  else
+    output.FileFormat() = a.FileFormat();
+
+  if (a.DataFormat() == format::unknown)
+  {
+    output.DataFormat() = b.DataFormat();
+    return output;
+  }
+  else if (b.DataFormat() == format::unknown)
+  {
+    output.DataFormat() = a.DataFormat();
+    return output;
+  }
+
+  if (a.DataFormat() != b.DataFormat())
+    throw std::runtime_error("Serialization formats don't match!");
+  else
+    output.DataFormat() = a.DataFormat();
+
+  return output;
+}
+
+class FatalOptions: public DataOptions
+{
+ public:
+  inline FatalOptions() : DataOptions() { this->Fatal() = true; }
```

Comment:
```suggestion
  inline FatalOptions() : DataOptions(), fatal(true) { }
```
Up to you, maybe this strategy seems a little cleaner?

You could also make them all `struct`s so that the `public:` isn't needed.